plugin.video.kla.tv/addon.py

```diff
@@ -0,0 +1,169 @@
+                      "langid": str(payload_id)
+                      }
+                 }
+
+    data = json.dumps(post_args)
+
+    content = urllib2.urlopen(req, data=data).read()
+
+    video_index = json.loads(content)
+
+    data_digest = video_index.get('data', [])
+    for video in data_digest:
+        title = video.get('title', '')
+        image = 'https://www.kla.tv/{}'.format(video.get('img', ''))
+        url = "https://www.kla.tv/{}".format(video.get('longlink', '')).replace('lang=de', 'lang={}'.format(lang_id))
+        if url == 'https://www.kla.tv/':
+            continue
+        addDir(title.encode('utf8'), url.encode('utf8'), 2, image)
+
+
+def VIDEOLINKS(url, name):
+    req = urllib2.Request(url)
+    for header in headers:
+        req.add_header(*header)
+
+    response = urllib2.urlopen(req)
+
+    link = response.read().replace('\n', '').replace('\t', ' ')
+    while '  ' in link:
+        link = link.replace('  ', ' ')
+    response.close()
+
+    compiled_regex = re.compile('file: "(.+?)", label: "(.+?)"')
+    value_tuples = compiled_regex.findall(link)
+
+    for entry in value_tuples:
+        if "blockid" in entry[0].lower() or "label:" in entry[0].lower():
+            continue
+        addLink('{0} ({1})'.format(name, entry[1]), 'https://www.kla.tv/{}'.format(entry[0]), __icon__)
+
+
+def get_params():
+    param = []
+    paramstring = sys.argv[2]
+    if len(paramstring) >= 2:
+        params = sys.argv[2]
+        cleanedparams = params.replace('?', '')
+        if (params[len(params) - 1] == '/'):
+            params = params[0:len(params) - 2]
+        pairsofparams = cleanedparams.split('&')
+        param = {}
+        for i in range(len(pairsofparams)):
+            splitparams = {}
+            splitparams = pairsofparams[i].split('=')
+            if (len(splitparams)) == 2:
+                param[splitparams[0]] = splitparams[1]
+
+    return param
+
+
+def addLink(name, url, iconimage):
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
+    liz.setInfo(type="Video", infoLabels={"Title": name})
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=liz)
+    return ok
+
+
+def addDir(name, url, mode, iconimage):
+    u = sys.argv[0] + "?url=" + urllib.quote_plus(url) + "&mode=" + str(mode) + "&name=" + urllib.quote_plus(name)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
+    liz.setInfo(type="Video", infoLabels={"Title": name})
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+    return ok
+
+
+params = get_params()
+url = None
+name = None
+mode = None
+
+try:
+    url = urllib.unquote_plus(params["url"])
+except:
+    pass
+try:
+    name = urllib.unquote_plus(params["name"])
+except:
+    pass
+try:
+    mode = int(params["mode"])
+except:
+    pass
+
+print "Mode: " + str(mode)
```

Comment:
Do not use print to log output. Please use xbmc.log() in its place.
Same in the many below this.